src/packages/media/media/components/input-upload-field/input-upload-field.element.ts

```diff
@@ -45,17 +48,39 @@ export class UmbInputUploadFieldElement extends UmbLitElement {
 	@state()
 	public temporaryFile?: UmbTemporaryFileModel;
 
+	public set src(src: string) {
+		this._src = src;
+		this._previewAlias = this.#getPreviewElementAlias();
+	}
+	public get src() {
+		return this._src;
+	}
+
 	@state()
 	private _src = '';
 
 	@state()
 	private _extensions?: string[];
 
+	@state()
+	private _previewAlias?: string;
+
 	@query('#dropzone')
 	private _dropzone?: UUIFileDropzoneElement;
 
 	#manager = new UmbTemporaryFileManager(this);
 
+	#previewers = new UmbArrayState(<Array<ManifestFileUploadPreview>>[], (x) => x.alias);
+
+	constructor() {
+		super();
+		new UmbExtensionsManifestInitializer(this, umbExtensionsRegistry, 'fileUploadPreview', null, (previews) => {
+			previews.forEach((preview) => {
+				this.#previewers.appendOne(preview.manifest);
+			});
```

Comment:
You will get the full updated list whenever you get an item here. So if you append one to the existing list, you will not remove any unregistered extensions. Have you considered storing the preview list in an array and overwriting each time?